version-ranges/src/lib.rs

```diff
@@ -1326,7 +1334,7 @@ pub mod tests {
 
         #[test]
         fn from_range_bounds(range in any::<(Bound<u32>, Bound<u32>)>(), version in version_strat()) {
-            let rv: Ranges<_> = Ranges::from_range_bounds(range);
+            let rv: Ranges<_> = Ranges::<u32>::from_range_bounds(range);
```

Comment:
I guess this means this is technically a breaking change. (Although we technically permit type inference regressions as semver compatible in Rust-library-land... Unless they are too disruptive.) I only mention this because I think `version-ranges` is published as a library?

Reply:
Agreed that this is technically a breaking changes, but i expect that real users use their Ranges as parameters to a function that has the generic parameter fix (the `DependencyProvider` has), so I consider it acceptable.